lib/AST/DistributedDecl.cpp

```diff
@@ -119,15 +119,8 @@ Type swift::getSerializationRequirementTypesForMember(
     }
 
     // Also store all `SerializationRequirement : SomeProtocol` requirements
-    for (auto requirement: signature.getRequirements()) {
-      if (requirement.getFirstType()->isEqual(SerReqAssocType) &&
-          requirement.getKind() == RequirementKind::Conformance) {
-        if (auto nominal = requirement.getSecondType()->getAnyNominal()) {
-          if (auto protocol = dyn_cast<ProtocolDecl>(nominal)) {
-            serializationRequirements.insert(protocol);
-          }
-        }
-      }
+    for (auto proto: signature->getRequiredProtocols(SerReqAssocType)) {
```

Comment:
Thank you so much for doing this along with the other cleanups. The code in this file is much easier to follow now!